### DISCUSSION TITLE
What is the recommended way to use `BIN_TO_UUID()` in Prisma?

### DISCUSSION THREAD

**Discussion by isaced** [prisma/prisma#22602] [Q&A]
### Question

What is the best way to use `binary UUID` in Prisma?

I'm currently using Prisma for developing my application and I need to convert binary data to UUID format. In my research, I came across the `BIN_TO_UUID()` function as a possible solution for this conversion within Prisma. However, I'm curious to know what the recommended way is to use `BIN_TO_UUID()` in Prisma.

**I know this should work, but is there a more convenient way to use it other than `$queryRaw`? I'd like to use `find()/findMany()/... `**

```js
const result = await prisma.$queryRaw`
  SELECT id, BIN_TO_UUID(binaryField) AS asUUID
  FROM myModel
`;
```

```
// schema.prisma
model User {
  id        Bytes   @id @default(dbgenerated("uuid_to_bin(uuid())")) @db.Binary(16)
  name      String
}
```

> ref: https://blogs.oracle.com/mysql/post/mysql-uuids

**Comment by nurul3101** [comment]
Hello @isaced 👋 

At the moment, I believe the most straightforward way is to use `$queryRaw`. Using [SafeQL](https://safeql.dev/compatibility/prisma.html) with Prisma would allow you to have type safety in raw queries, which should make the experience better.

I was able to find a related GitHub Issue about using `BIN_TO_UUID()` in the `dbgenerated()` function, but it looks like it was resulting into an error.

- https://github.com/prisma/prisma/issues/15727#issuecomment-1284624850

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
